model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/DiffView.kt

```diff
@@ -70,10 +71,11 @@ class DiffView(private val repositoryManager: RepositoriesManager) {
         application.routing {
             requiresLogin {
                 get("/diff") {
-                    val visibleRepositories = repositoryManager.getRepositories().filter {
-                        call.hasPermission(ModelServerPermissionSchema.repository(it).list)
-                    }
+                    @OptIn(RequiresTransaction::class)
```

Comment:
`@OptIn(RequiresTransaction::class)` used,  but no transaction started.


Resulting in:

```
org.modelix.model.server.store.MissingTransactionException: Transaction required
	at org.modelix.model.server.store.TransactionLocksKt.assertRead(TransactionLocks.kt:56)
	at org.modelix.model.server.store.IgniteStoreClient.getAll(IgniteStoreClient.kt:105)
	at org.modelix.model.server.store.IGenericStoreClient.getAll(IGenericStoreClient.kt:35)
	at org.modelix.model.server.store.IGenericStoreClient.get(IGenericStoreClient.kt:31)
	at org.modelix.model.server.handlers.RepositoriesManager.getRepositories(RepositoriesManager.kt:122)
	at org.modelix.model.server.handlers.RepositoriesManager.getRepositories(RepositoriesManager.kt:117)
	at org.modelix.model.server.handlers.ui.DiffView$init$1$1$1.invokeSuspend$lambda$2(DiffView.kt:76)
	at io.ktor.server.html.RespondHtmlTemplateKt.respondHtmlTemplate(RespondHtmlTemplate.kt:20)
	at io.ktor.server.html.RespondHtmlTemplateKt.respondHtmlTemplate$default(RespondHtmlTemplate.kt:15)
	at org.modelix.model.server.handlers.ui.DiffView$init$1$1$1.invokeSuspend(DiffView.kt:75)
```

Reply:
Fixed by using `respondHtmlTemplateInTransaction`.

I first introduced `respondHtmlTemplateInTransaction` and later `@RequiresTransaction` and then reverted all `respondHtmlTemplateInTransaction` calls that din't need a transaction. Here I missed that there is the `@OptIn` while reverting it.

The solution with the annotations is still not perfect, but the best we have.